### PR TITLE
`Development`: Speedup client tests for text submission component

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [ ] * I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [ ] * Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [ ] * I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [ ] * I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] * I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [ ] * I implemented the changes with a good performance and prevented too many database calls.
- [ ] * I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [ ] * I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ] * I documented the TypeScript code using JSDoc style.
- [ ] * I added multiple screenshots/screencasts of my UI changes.
- [ ] * I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] * I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] * I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

*not appicable
### Motivation and Context

This PR reduces the time necessary to run the test file `text-submission-viewer.component.spec.ts`

### Description

I removed `ArtemisPlagiarismModule`, mocked `SplitPaneHeaderComponent` and declared the component that is tested in this file.

### Steps for Testing

Mainly code review

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage

text-submission-viewer.component.ts: 65.38% | 92.98%

### CODE REVIEW

File: src/test/javascript/spec/component/plagiarism/text-submission-viewer.component.spec.ts [hunk 1]
```diff
@@ -33,7 +34,8 @@ describe('Text Submission Viewer Component', () => {
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [ArtemisTestModule, ArtemisPlagiarismModule, TranslateTestingModule],
+            imports: [ArtemisTestModule, TranslateTestingModule],
+            declarations: [TextSubmissionViewerComponent, MockComponent(SplitPaneHeaderComponent)],
```
Comment (b-fein) [inline src/test/javascript/spec/component/plagiarism/text-submission-viewer.component.spec.ts:38]:
```suggestion
            imports: [ArtemisTestModule],
            declarations: [TextSubmissionViewerComponent, MockComponent(SplitPaneHeaderComponent), MockPipe(ArtemisTranslatePipe)],
```
I think you can remove the import if you mock the translation pipe.

**Review by b-fein — APPROVED** [review]
Looks good. :+1: 
I found one place where I think another small improvement could be made.

File: src/test/javascript/spec/component/plagiarism/text-submission-viewer.component.spec.ts (hunk 1 above)
Comment (willergeorg) [reply src/test/javascript/spec/component/plagiarism/text-submission-viewer.component.spec.ts]: Thanks for the hint, done in bed923d

**Review by b-fein — APPROVED** [review]

**Review by cremertim — APPROVED** [review]
code changes lgtm

**Pull request merged**